js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/ExecuteChecksButton.tsx

```diff
@@ -63,8 +63,17 @@ export const ExecuteChecksButton = ({
     );
   }
 
-  const jobName = jobNames[0];
-  if (!jobName) {
+  const commonCheckJobNames = checks.reduce((commonJobNames, check) => {
+    return commonJobNames.filter((name: string) => check.jobNames.includes(name));
+  }, checks[0]?.jobNames || []);
```

Comment:
I'm finding this code fairly hard to understand and I wonder if there's a clearer way to write it. What is it aiming to do?

Reply:
Each `check` has a list `jobNames`, and this code finds the set of job names that are common amongst all. This approach takes the first set of `jobNames` and then reduces across the rest of the checks filtering the set of names to those that appear in each check. I asked an LLM and this was the recommendation that seemed cleanest to me. Agree its not the most legible, happy to change if you have something cleaner in mind.

Reply:
Hmm - something like `intersection(commonJobNames, ...checks.map(c => c.jobNames))` (using the lodash helper library) might be equivalent and a bit cleaner since "intersection" handles the dirty work, but I think this is also nice as pure JS